cleverpush/src/main/java/com/cleverpush/CleverPush.java

```diff
@@ -2606,15 +2611,25 @@ public void showTopicsDialog(Activity dialogActivity, TopicsDialogListener topic
 
         checkboxLayout.addView(scrollView);
 
-        AlertDialog.Builder alertBuilder =
-            getTopicAlertBuilder(dialogActivity, topicsDialogListener, themeResId, channelConfig, channelTopics,
-                hasDeSelectAllInitial, checkboxLayout, unsubscribeCheckbox, selectedTopics);
-        AlertDialog alert = alertBuilder.create();
-        alert.setOnShowListener(dialog -> {
-          Logger.d(LOG_TAG, "showTopicsDialog activity: " + dialogActivity.getClass().getCanonicalName());
-          showingTopicsDialog = true;
-        });
-        alert.show();
+        try {
+          AlertDialog.Builder alertBuilder =
+                  getTopicAlertBuilder(dialogActivity, topicsDialogListener, themeResId, channelConfig, channelTopics,
+                          hasDeSelectAllInitial, checkboxLayout, unsubscribeCheckbox, selectedTopics);
+          AlertDialog alert = alertBuilder.create();
+          alert.setOnShowListener(dialog -> {
+            Logger.d(LOG_TAG, "showTopicsDialog activity: " + dialogActivity.getClass().getCanonicalName());
+            showingTopicsDialog = true;
```

Comment:
wont this stop the recursive call from showing the dialog? with L2562 isShowingTopicsDialog